std/meta.d

```diff
@@ -1309,6 +1309,62 @@ private template staticMerge(alias cmp, int half, Seq...)
     }
 }
 
+/**
+ * Allows subsetting of alias sequence that are not
+ * necessarily adjacent to each other, i.e. cannot be sliced.
+ *
+ * Parameters:
+ *	range = a range object preferably an unsigned array to index T
+ *  T = a set of types to be indexed
+ *
+ * Returns: alias sequence of selected types
+ *
+ * Evaluates to `AliasSeq!(T[range[0]], T[range[1]], ..., T[range[n]])`.
+ *
+ * How:
+ * 		alias T1 = staticSubset!([0, 2, 1, 1, 2, 0], int, char, double);
+ */
+template staticSubset(alias range, T...)
+{
+	import std.range : isInputRange;
+	import std.traits : isArray, isNarrowString;
+
+	alias ArrT = typeof(range);
+	static if (isArray!ArrT && !isNarrowString!ArrT)
+	{
+		static if (range.length == 0)
+		{
+			alias staticSubset = AliasSeq!();
+		}
+		else
+		{
+			alias staticSubset = AliasSeq!(T[range[0]], staticSubset!(range[1 .. $], T));
+		}
+	}
+	else static if (isInputRange!ArrT)
```

Comment:
I shall amend the import to import `std.range.primitives : isInputRange;` however ...

The else block operates at the function level, doesn't that mean that placing the import for the conditional statement at the top of the else block inside the block itself will cause a bug?

```
else static if (isInputRange!ArrT)
	{
		import std.range.primitives : isInputRange; // import included after call
		import std.array : array;
		alias staticSubset = staticSubset!(array(range));
	}
```
The only other else blocks in the function will not flow through the `isInputRange` call. The `isInputRange` call is at the template level so does the import call not have to be made at that point?